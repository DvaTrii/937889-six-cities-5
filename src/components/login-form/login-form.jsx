import React, {createRef, PureComponent} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {login} from "../../store/user/actions";

class LoginForm extends PureComponent {
  constructor(props) {
    super(props);

    this.loginRef = createRef();
    this.passwordRef = createRef();
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(evt) {
    evt.preventDefault();

    this.props.onSubmit({
      login: this.loginRef.current.value,
      password: this.passwordRef.current.value,
    });
  }

  render() {
    return (
      <form
        className="login__form form"
        action="#"
        method="post"
        onSubmit={this.handleSubmit}>
        <div className="login__input-wrapper form__input-wrapper">
          <label className="visually-hidden">E-mail</label>
          <input
            ref={this.loginRef}
            className="login__input form__input"
            type="email"
            name="email"
            placeholder="Email"
            required=""/>
        </div>
        <div className="login__input-wrapper form__input-wrapper">
          <label className="visually-hidden">Password</label>
          <input
            ref={this.passwordRef}
            className="login__input form__input"
            type="password"
            name="password"
            placeholder="Password"
            required="" />
        </div>
        <button className="login__submit form__submit button" type="submit">Sign in</button>
      </form>
    );
  }
}

LoginForm.propTypes = {
  onSubmit: PropTypes.func.isRequired
};

const mapDispatchToProps = (dispatch) => ({
  onSubmit(authData) {
    dispatch(login(authData));
  }
});

export {LoginForm};
export default connect(null, mapDispatchToProps)(LoginForm);
