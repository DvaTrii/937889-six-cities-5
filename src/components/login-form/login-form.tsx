import * as React from "react";
import {connect} from "react-redux";
import {login} from "../../store/user/operations";

type Props = {
  onSubmit: ({login, password}: { login: string; password: string }) => void;
};

const LoginForm: React.FC<Props> = ({onSubmit}) => {

  const loginRef = React.useRef<HTMLInputElement>(null);
  const passwordRef = React.useRef<HTMLInputElement>(null);

  const handleSubmit = (evt) => {
    evt.preventDefault();

    onSubmit({
      login: loginRef.current.value,
      password: passwordRef.current.value,
    });
  };

  return (
    <form
      className="login__form form"
      action="#"
      method="post"
      onSubmit={handleSubmit}>
      <div className="login__input-wrapper form__input-wrapper">
        <label className="visually-hidden">E-mail</label>
        <input
          ref={loginRef}
          className="login__input form__input"
          type="email"
          name="email"
          placeholder="Email"
        />
      </div>
      <div className="login__input-wrapper form__input-wrapper">
        <label className="visually-hidden">Password</label>
        <input
          ref={passwordRef}
          className="login__input form__input"
          type="password"
          name="password"
          placeholder="Password"
        />
      </div>
      <button className="login__submit form__submit button" type="submit">Sign in</button>
    </form>
  );

};

const mapDispatchToProps = (dispatch) => ({
  onSubmit(authData) {
    dispatch(login(authData));
  }
});

export {LoginForm};
export default connect(null, mapDispatchToProps)(LoginForm);
