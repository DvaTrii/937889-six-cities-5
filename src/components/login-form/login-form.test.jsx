import React from "react";
import renderer from "react-test-renderer";
import {LoginForm} from "./login-form";

it(`Should LoginForm render properly`, () => {

  const tree = renderer
    .create(
        <LoginForm
          onSubmit={() => {}}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
