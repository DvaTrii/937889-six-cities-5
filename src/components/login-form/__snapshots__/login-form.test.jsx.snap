// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should LoginForm render properly 1`] = `
<form
  action="#"
  className="login__form form"
  method="post"
  onSubmit={[Function]}
>
  <div
    className="login__input-wrapper form__input-wrapper"
  >
    <label
      className="visually-hidden"
    >
      E-mail
    </label>
    <input
      className="login__input form__input"
      name="email"
      placeholder="Email"
      required=""
      type="email"
    />
  </div>
  <div
    className="login__input-wrapper form__input-wrapper"
  >
    <label
      className="visually-hidden"
    >
      Password
    </label>
    <input
      className="login__input form__input"
      name="password"
      placeholder="Password"
      required=""
      type="password"
    />
  </div>
  <button
    className="login__submit form__submit button"
    type="submit"
  >
    Sign in
  </button>
</form>
`;
